libs/components/src/next/form/BaseFormControl.ts

```diff
@@ -0,0 +1,135 @@
+import { PropertyDeclarations } from 'lit';
+import {
+    FormControlMixin,
+    maxLengthValidator,
+    minLengthValidator,
+    patternValidator,
+    requiredValidator,
+} from '@open-wc/form-control';
+import { LitElement } from 'lit';
+import { submit } from '@open-wc/form-helpers';
+import { maxValueValidator, minValueValidator } from './validators';
+import { ERROR_MESSAGE_CUSTOM_TAG } from './error-message/vl-error-message.component';
+import { BaseLitElement } from '@domg-wc/common-utilities';
+
+export abstract class BaseFormControl extends FormControlMixin(BaseLitElement) {
+    id = '';
+    protected name = '';
+    protected label = '';
+    protected block = false;
+    protected required = false;
+    protected isInvalid = false;
+
+    static formControlValidators = [
+        requiredValidator,
+        minLengthValidator,
+        maxLengthValidator,
+        minValueValidator,
+        maxValueValidator,
+        patternValidator,
+    ];
+
+    static shadowRootOptions = { ...LitElement.shadowRootOptions, delegatesFocus: true };
+
+    static get properties(): PropertyDeclarations {
+        return {
+            id: { type: String, reflect: true },
+            name: { type: String, reflect: true },
+            label: { type: String, reflect: true },
+            block: { type: Boolean, reflect: true },
```

Comment:
weet dat we dit hadden besproken om later te doen; maar mss toch al overwegen eventueel `disabled: { type: Boolean, reflect: true },` nog toe te voegen?